e2e/infra/terraform/Makefile

```diff
@@ -0,0 +1,325 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+.ONESHELL:
+.SHELL := /usr/bin/bash
+.PHONY: help validate up_vars down_vars apply apply-auto-approve decode destroy-backend destroy destroy-target import plan-destroy plan plan-target
+VARS="variables/$(ENV)-$(TIER).tfvars"
+REGION?=us-east-1
+S3_BUCKET="$(ENV)-${COMPANY}-$(REGION)-terraform-state"
+DYNAMODB_TABLE="$(ENV)-$(TIER)-${COMPANY}-$(REGION)-terraform"
+#Include directory to have multiple workspaces in the same repo
+DIRECTORY=$(shell basename "$$(pwd)")
+WORKSPACE="$(ENV)-$(TIER)-$(REGION)-$(DIRECTORY)"
+COMPANY=fingerprint
+BOLD=$(shell tput -T xterm bold)
+RED=$(shell tput -T xterm setaf 1)
+GREEN=$(shell tput -T xterm setaf 2)
+YELLOW=$(shell tput -T xterm setaf 3)
+RESET=$(shell tput -T xterm sgr0)
+PAR=200
+EXTRA_ARGS?=
+
+CURRENT_FOLDER=$(shell git remote get-url --push origin | awk -F'/' '{print $$5}' | awk -F'.' '{print $$1}')
+BUCKET_KEY=$(shell git remote get-url --push origin | awk -F'/' '{print $$5}' | awk -F'.' '{print $$1}')
+
+ifeq ($(ENV)$(REGION),devus-east-1)
+AWS_PROFILE?=fpjs_dev
+ACCOUNT=708050157146
+KEY_ARN=1fd4b7e6-1547-4b62-859e-469aa2c0ca83
+else ifeq ($(ENV)$(REGION),releaseus-east-1)
+AWS_PROFILE?=fpjs_dev
+ACCOUNT=708050157146
+KEY_ARN=1fd4b7e6-1547-4b62-859e-469aa2c0ca83
+endif
+
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+set-env:
+	@if [ -z $(ENV) ]; then \
+		echo "$(BOLD)$(RED)ENV was not set$(RESET)"; \
+		ERROR=1; \
+	 fi
+	@if [ -z $(REGION) ]; then \
+		echo "$(BOLD)$(RED)REGION was not set$(RESET)"; \
+		ERROR=1; \
+	 fi
+	@if [ ! -z $${ERROR} ] && [ $${ERROR} -eq 1 ]; then \
+		echo "$(BOLD)Example usage: \`ENV=demo REGION=us-east-2 make plan\`$(RESET)"; \
+		exit 1; \
+	 fi
+	@if [ ! -f "$(VARS)" ]; then \
+		echo "$(BOLD)$(RED)Could not find variables file: $(VARS)$(RESET)"; \
+		if ! aws --profile $(AWS_PROFILE) s3api head-object --bucket $(ENV)-${COMPANY}-$(REGION)-terraform-state --key "$(CURRENT_FOLDER)/$(ENV)-$(TIER).tfvars"  > /dev/null 2>&1 ; then
+			echo "\n$(ENV)-$(TIER).tfvars not found on $(ENV)-${COMPANY}-$(REGION)-terraform-state/$(CURRENT_FOLDER)\n"; \
```

Comment:
nit: why repeat `$(ENV)-$(TIER).tfvars` when we can just `$(VARS)`? This is in multiple places. Could lead to incorrect logging or buggy behavior when these two values become out-of-sync